CMakeLists.txt

```diff
@@ -167,7 +167,6 @@ ELSE()
   ELSE()
     ADD_NONDEBUG_COMPILE_FLAG("-O2")
   ENDIF()
-  ADD_NONDEBUG_COMPILE_FLAG("-UNDEBUG") # Keep asserts.
```

Comment:
doesn't your change remove asserts?

NDEBUG means "no debug"; undefining it means debug; removing this line means no debug...

or did I get mixed up? :)

Reply:
Or was this intentional, and you don't think release builds should have assertions?

Reply:
Yes this line mean that assert were always on even in release builds.

The idea is to strip them out in release builds.

Reply:
cmake itself explistly adds `-DNDEBUG` in release builds.   This line mean that out release build currently build with `-DNDEBUG -UNDEBUG`.. which is odd, and goes against tradition.

I think maybe it was done so that WASM_UNREACHABLE showed useful information even in release build, but I think we can achieve that in other ways.

Reply:
Oh, historically I believe the reason was: debug builds are too slow to ship to users, but binaryen and wasm are new and surely we have lots of bugs, so lets ship release+asserts builds to our users, to catch more bugs early.

I think it makes sense to continue with that, but if you disagree, then I think it would be good to at minimum add a testing mode in binaryen CI that enables assertions, and preferably to also test with assertions on in emscripten CI - but that's a little risky as it's not the build that users get :(

Reply:
Ok, all add some comments there, and maybe add a separate Relase+Asserts config.